here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/WriteFeatureApiTask.java

```diff
@@ -228,6 +238,202 @@ protected void init() {}
+          }
+          final String featureIdFromErr = matcherFeatureId.group(1);
+          // Find the requested change for the corresponding feature with that ID
+          for (XyzFeature requestedChange : featuresFromRequest) {
+            if (requestedChange.getId().equals(featureIdFromErr)) {
+              // If UUID input by user, return conflict
+              if (requestedChange
+                      .getProperties()
+                      .getXyzNamespace()
+                      .getUuid()
+                  != null) {
+                return verticle.sendErrorResponse(
+                    routingContext,
+                    XyzError.CONFLICT,
+                    "Error updating feature '" + featureIdFromErr + "', wrong UUID.");
+              }
+              // Else the feature was modified concurrently within Naksha
+              if (retry >= MAX_RETRY_ATTEMPT) {
+                logger.error(
+                    "Max retry attempt for PATCH REST API reached, too many concurrent modification, error: {}",
+                    er.message);
+                return verticle.sendErrorResponse(
+                    routingContext,
+                    XyzError.EXCEPTION,
+                    "Max retry attempt for PATCH REST API reached, too many concurrent modification, error: "
+                        + er.message);
+              }
+              // Attempt retry
+              return attemptFeaturesPatching(
+                  spaceId, featureIds, featuresFromRequest, responseType, retry + 1);
+            }
+          }
+        } else {
+          // Other types of error, convert result to ErrorResponse
+          logger.error("Received error result {}", er);
+          return verticle.sendErrorResponse(routingContext, er.reason, er.message);
+        }
+      } else {
+        try {
+          final Map<EExecutedOp, List<XyzFeature>> featureMap =
+              readFeaturesGroupedByOp(wrResult, XyzFeature.class);
+          final List<XyzFeature> insertedFeatures = featureMap.get(EExecutedOp.CREATED);
+          final List<XyzFeature> updatedFeatures = featureMap.get(EExecutedOp.UPDATED);
+          return verticle.sendXyzResponse(
+              routingContext,
+              responseType,
+              new XyzFeatureCollection()
+                  .withInsertedFeatures(insertedFeatures)
+                  .withUpdatedFeatures(updatedFeatures));
+        } catch (NoCursor | NoSuchElementException emptyException) {
+          return verticle.sendErrorResponse(
+              routingContext, XyzError.EXCEPTION, "Unexpected empty result from ResultCursor");
+        }
+      }
+    }
+    return verticle.sendErrorResponse(
+        routingContext,
+        XyzError.EXCEPTION,
+        "Unexpected code point reached while attempting to PATCH features: " + featureIds);
+  }
+
+  private Difference removeAllRemoveOp(Difference difference) {
```

Comment:
It seems this function is in Test file as well. Let's move it to some common file, which can be referred at multiple places.

Reply:
I moved it to a util class in the patcher module in lib-core